core/src/main/java/org/apache/iceberg/ManifestsTable.java

```diff
@@ -73,10 +71,15 @@ MetadataTableType metadataTableType() {
   protected DataTask task(TableScan scan) {
     TableOperations ops = operations();
     String location = scan.snapshot().manifestListLocation();
+    Map<Integer, PartitionSpec> specs = table().specs();
```

Comment:
Can you update this to copy the map into a new `HashMap`? Not all maps are serializable with Kryo, so we should make sure it is created by `Maps.newHashMap` to be safe.

Reply:
updated